src/containers/gui.jsx

```diff
@@ -52,6 +54,40 @@ class GUI extends React.Component {
         setIsScratchDesktop(this.props.isScratchDesktop);
         this.setReduxTitle(this.props.projectTitle);
         this.props.onStorageInit(storage);
+
+        // Use setTimeout. Do not use requestAnimationFrame or a resolved
+        // Promise. We want this work delayed until after the data request is
+        // made.
+        setTimeout(() => {
```

Comment:
Crud. I fixed this yesterday and forgot to push the change.

The setTimeout doesn't create a race condition. The race condition is effectively in render() since its what blocks the existing renderer creation. We need to test for this in the if block before rendering the GUI.

Here's the fixed logic.

```js
if (
    !fontsLoaded ||
    fetchingProject ||
    this.props.vm.renderer && isLoading
) {
    // TODO: Render a loading screen.
    return null;
}
```

I'm going to take this a step further and ensure the renderer by this point.